### PR TITLE
Split aio.rs to separate files.

### PR DESCRIPTION
**Pull request by nihohit**
This change doesn't modify any code, except for `use` and `mod` statements.

### CODE REVIEW

**Comment by jaymell** [comment]
If you can rebase/update this, I'd like to get it in :pray:

**Comment by nihohit** [comment]
@jaymell rebased

File: redis/src/lib.rs [hunk 1]
```diff
@@ -415,6 +415,7 @@ mod pipeline;
 #[cfg_attr(docsrs, doc(cfg(feature = "acl")))]
 pub mod acl;
 
+/// module for async support.
```
Comment (jaymell) [inline redis/src/lib.rs:418]: Can we put the existing doc comment at the top of aio/mod.rs? (Though maybe the line "experimental" doesn't need to be in there anymore)

**Comment by nihohit** [comment]
Works to like me to move async_cluster also to the aio module?

**Comment by jaymell** [comment]
> Works to like me to move async_cluster also to the aio module?

I think for now just breaking up `aio.rs` is good. Maybe the various cluster modules should be consolidated in some way, but honestly I'm not that concerned about it.

File: redis/src/lib.rs (hunk 1 above)
Comment (nihohit) [reply redis/src/lib.rs:418]: done

Comment (jaymell) [reply redis/src/lib.rs:418]: I think we need to remove this comment though, else it will break the module description in the top-level docs.

Comment (nihohit) [reply redis/src/lib.rs]: done

**Review by jaymell — APPROVED** [review]

**Comment by jaymell** [comment]
Thanks!

**Pull request merged**